src/scripts/main.js

```diff
@@ -1,3 +1,578 @@
+    } else if (value <= 15) {
+      arrayCells[3][val] = 2;
+    }
+  }
+
+  if (num === 4) {
+    if (value <= 3) {
+      arrayCells[0][val] = 4;
+    } else if (value <= 7) {
+      arrayCells[1][val] = 4;
+    } else if (value <= 11) {
+      arrayCells[2][val] = 4;
+    } else if (value <= 15) {
+      arrayCells[3][val] = 4;
+    }
+  }
+}
+
+function cleaner() {
+  cells.forEach(cell => {
+    cell.textContent = '';
+
+    cellToRemove.forEach(Name => {
+      cell.classList.remove(Name);
+    });
+  });
+
+  arrayCells = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  score = 0;
+  scoreBoard.textContent = `${score}`;
+}
+
+function refresh() {
+  for (let i = 0; i < arrayClassCells.length; i++) {
+    for (let j = 0; j < arrayClassCells[i].length; j++) {
+      arrayClassCells[i][j] = `field_cell--${arrayCells[i][j]}`;
+
+      const value = i * 4 + j;
+
+      if (arrayCells[i][j] !== 0) {
+        cells[value].classList.add(arrayClassCells[i][j]);
+        cells[value].textContent = `${arrayCells[i][j]}`;
+      }
+
+      if (arrayCells[i][j] === 0) {
+        cellToRemove.forEach(Name => {
+          cells[value].classList.remove(Name);
+        });
+        cells[value].textContent = ``;
+      }
+    }
+  }
+}
+
+const arrayClassCells = [
+  [
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+  ],
+  [
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+  ],
+  [
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+  ],
+  [
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+    `field_cell--${0}`,
+  ],
+];
+
+function victory() {
+  for (let i = 0; i < arrayCells.length; i++) {
+    for (let j = 0; j < arrayCells[i].length; j++) {
+      if (arrayCells[i][j] === 2048) {
+        messageWin.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+gameBoard.addEventListener('keydown', function(move) {
+  if (
+    start.classList.contains('restart')
+    && messageWin.classList.contains('hidden')
+  ) {
+    const arrayCellsOld = JSON.parse(JSON.stringify(arrayCells));
+    let result = false;
+
+    if (move.key === 'ArrowUp') {
+      moveUp();
+      refresh();
+    }
+
+    if (move.key === 'ArrowDown') {
+      moveDown();
+      refresh();
+    }
+
+    if (move.key === 'ArrowRight') {
+      moveRight();
+      refresh();
+    }
+
+    if (move.key === 'ArrowLeft') {
+      moveLeft();
+      refresh();
+    }
+
+    victory();
+
+    for (let i = 0; i < arrayCells.length; i++) {
+      for (let j = 0; j < arrayCells[i].length; j++) {
+        if (arrayCells[i][j] !== arrayCellsOld[i][j]) {
+          result = true;
+        }
+      }
+    }
+
+    if (result) {
+      addRandomValues();
+    }
+
+    scoreBoard.textContent = `${score}`;
+  }
+});
+
+function moveUp() {
+  for (let j = 3; j >= 0; j--) {
```

Comment:
The function 'moveUp' has complex nested loops and conditionals. Consider breaking it down into smaller functions for better readability and maintainability.